parm/config/config.com

```diff
@@ -0,0 +1,91 @@
+
+# These are just templates. All templates must use single quotations so variable
+#   expansion does not occur when this file is sourced. Substitution happens later
+#   during runtime. It is recommended to use the helper function `generate_com()`,
+#   to do this substitution, which is defined in `ush/preamble.sh`.
+#   
+#   Syntax for generate_com():
+#       generate_com [-rx] $var1[:$tmpl1] [$var2[:$tmpl2]] [...]]
+#
+#       options:
+#           -r: Make variable read-only (same as `decalre -r`)
+#           -x: Mark variable for declare -rx (same as `declare -x`)
+#       var1, var2, etc: Variable names whose values will be generated from a template
+#                   and declared
+#       tmpl1, tmpl2, etc: Specify the template to use (default is "${var}_TMPL")
+#
+#   Examples:
+#       # Current cycle and RUN
+#       YMD=${PDY} HH=${cyc} generate_com -rx COM_ATMOS_ANALYSIS
+#
+#       # Previous cycle and gdas
+#       RUN=${GDUMP} YMD=${gPDY} HH=${gcyc} generate_com -rx \
+#           COM_ATMOS_HISTORY_PREV:COM_ATMOS_HISTORY_TMPL
+#
+#       # Current cycle and COM for first member
+#       MEMDIR='mem001' YMD=${PDY} HH=${cyc} generate_com -rx COM_ATMOS_HISTORY
+#
+
+# 
+# If any restart, input, or analysis template is updated, `setup_expt.py.fill_COMROT_cycled()`
+#   must correspondingly be updated to match.
+# 
+if [[ "${RUN_ENVIR:-emc}" == "nco" ]]; then
+    COM_OBS_TMPL=$(compath.py "${envir}/obsproc/${obsproc_ver}")'/${RUN}.${YMD}/${HH}/atmos'
+    COM_RTOFS_TMPL=$(compath.py "${envir}/${WAVECUR_DID}/${rtofs_ver}")
+else
+    COM_OBS_TMPL='${ROTDIR}/${RUN}.${YMD}/${HH}/obs'
+    COM_RTOFS_TMPL='${DMPDIR}'
+fi
+declare -rx COM_OBS_TMPL COM_RTOFS_TMPL
+declare -rx COM_OBSDMP_TMPL='${DMPDIR}/${DUMP}${DUMP_SUFFIX}.${YMD}/${HH}/atmos'
+
+COM_BASE='${ROTDIR}/${RUN}.${YMD}/${HH}/${MEMDIR}'
+
+declare -rx COM_TOP_TMPL='${ROTDIR}/${RUN}.${YMD}/${HH}'
+
+declare -rx COM_ATMOS_INPUT_TMPL=${COM_BASE}'/model_data/atmos/input'
+declare -rx COM_ATMOS_RESTART_TMPL=${COM_BASE}'/model_data/atmos/restart'
+declare -rx COM_ATMOS_ANALYSIS_TMPL=${COM_BASE}'/analysis/atmos'
+declare -rx COM_ATMOS_HISTORY_TMPL=${COM_BASE}'/model_data/atmos/history'
+declare -rx COM_ATMOS_MASTER_TMPL=${COM_BASE}'/model_data/atmos/master'
+declare -rx COM_ATMOS_GRIB_TMPL=${COM_BASE}'/products/atmos/grib/${RES}'
+declare -rx COM_ATMOS_BUFR_TMPL=${COM_BASE}'/products/atmos/bufr'
+declare -rx COM_ATMOS_GEMPAK_TMPL=${COM_BASE}'/products/atmos/gempak'
+declare -rx COM_ATMOS_GENESIS_TMPL=${COM_BASE}'/products/atmos/cyclone/genesis_vital'
+declare -rx COM_ATMOS_TRACK_TMPL=${COM_BASE}'/products/atmos/cyclone/tracks'
+declare -rx COM_ATMOS_GOES_TMPL=${COM_BASE}'/products/atmos/goes_sim'
+declare -rx COM_ATMOS_IMAGERY_TMPL=${COM_BASE}'/products/atmos/imagery'
+declare -rx COM_ATMOS_MINMON_TMPL=${COM_BASE}'/products/atmos/minmon'
+declare -rx COM_ATMOS_WAFS_TMPL=${COM_BASE}'/products/atmos/wafs'
+declare -rx COM_ATMOS_WMO_TMPL=${COM_BASE}'/products/atmos/wmo'
```

Comment:
Potentially nitpicky comment...is there an order to these? If not, can these lines in each block be ordered alphabetically (e.g. start with the `COM_ATMOS_ANALYSIS_TMPL` line)? Ignore my comment if these are already ordered.

Reply:
I think they are in the order they are produced (or close to it). But they don't have to be.

Reply:
Ah, ok. :)